addons/account/models/account_bank_statement.py

```diff
@@ -104,6 +105,11 @@ class AccountBankStatement(models.Model):
     # -------------------------------------------------------------------------
     # COMPUTE METHODS
     # -------------------------------------------------------------------------
+    @api.depends('create_date')
+    def _compute_name(self):
+        for stmt in self:
+            stmt.name = _("%s Statement %s", stmt.journal_id.code, stmt.date)
```

Comment:
Are you not loosing the name when saving since `create_date` is set upon the creation

Reply:
Nope, it works as expected